### PR TITLE
Adds ability for exception modules to have any special characters as long as the delimiter is correct

### PR DESCRIPTION
**Pull request by bwarren2**
I have a Django project with a root directory that has a `-`, and I want to constrain imports between apps.  This works fine, but I can't add exceptions because they have the root directory in them, and the regex disallows `-`.  I have also tried evaluating the project from within using a root directory of `.` or not including root directory to no avail.

This MR allows for a root directory with a `-` to have import_ignores set for it.

### CODE REVIEW

File: src/importlinter/domain/fields.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ class DirectImportField(Field):
     Expects raw data in the form: "mypackage.foo.importer -> mypackage.bar.imported".
     """
 
-    DIRECT_IMPORT_STRING_REGEX = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")
+    DIRECT_IMPORT_STRING_REGEX = re.compile(r"^([\w\.-]+) -> ([\w\.-]+)$")
```
Comment (kingbuzzman) [inline src/importlinter/domain/fields.py:120]: I would do:

```
class DirectImportField(Field):
    """
    A field for DirectImports.
    Expects raw data in the form: "mypackage.foo.importer -> mypackage.bar.imported".
    """
    def parse(self, raw_data: Union[str, List]) -> DirectImport:
        string = StringField().parse(raw_data)
        importer, _, imported = string.partition(' -> ')
        if not (importer and imported):
            raise ValidationError('Must be in the form "package.importer -> package.imported".')
        return DirectImport(importer=Module(importer), imported=Module(imported))
```

Instead, and avoid the `re` altogether!

**Review by kingbuzzman — CHANGES_REQUESTED** [review]

**Comment by kingbuzzman** [comment]
Ps. the tests are failing, https://travis-ci.org/github/seddonym/import-linter/jobs/727507116

**Comment by seddonym** [comment]
> Ps. the tests are failing, https://travis-ci.org/github/seddonym/import-linter/jobs/727507116

Actually this isn't to do with this PR - this is a problem on master. Thanks for drawing my attention!

File: src/importlinter/domain/fields.py (hunk 1 above)
Comment (seddonym) [reply src/importlinter/domain/fields.py:120]: I like both ways, don't feel strongly.

**Review by seddonym — CHANGES_REQUESTED** [review]
Thanks for this! Looks good, but I'd like to see a test that covers this. Let me know if you need any pointers.

PS I've noticed there is a problem with the build on master, I'm looking into this separately.

**Comment by seddonym** [comment]
Ok master is now working again, as is Travis CI integration.

**Comment by bwarren2** [comment]
Changes made!

**Review by kingbuzzman — APPROVED** [review]
Looks good!

File: src/importlinter/domain/fields.py (hunk 1 above)
Comment (kingbuzzman) [reply src/importlinter/domain/fields.py]:
```
>>> import timeit
>>> timeit.timeit('x.match("hello -> yellow").groups()', 'import re; x = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")')
0.6223237819999952
>>> timeit.timeit('x.match("hello -> yellow").groups()', 'import re; x = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")')
0.6110563539999987
>>> timeit.timeit('x.match("hello -> yellow").groups()', 'import re; x = re.compile(r"^([\w\.]+) -> ([\w\.]+)$")')
0.6207327580000026
>>> timeit.timeit('"hello -> yellow".partition(" -> ")')
0.1567364259999806
>>> timeit.timeit('"hello -> yellow".partition(" -> ")')
0.15910449500000823
>>> timeit.timeit('"hello -> yellow".partition(" -> ")')
0.15209861200000319
```

I like regex -- but i only use it when it's something complex.. this doesn't warren it.

**Comment by kingbuzzman** [comment]
@seddonym what kind of time frame are looking at to see this merged and released?

**Review by seddonym — APPROVED** [review]

**Comment by seddonym** [comment]
> @seddonym what kind of time frame are looking at to see this merged and released?

It's on master now so hopefully that unblocks you. But I'll try to do a new release over the next few days.

**Comment by seddonym** [comment]
Thanks for everyone's help 😄

**Comment by kingbuzzman** [comment]
Yeah that helps, we need just need a pip version, but this is a step in the right direction. I appreciate your time, and your project ;)

**Comment by seddonym** [comment]
@kingbuzzman Latest version (1.2) of import-linter now released, enjoy.

**Pull request merged**